profile/src/wallet_kit_common/utils/logged_panic.rs

```diff
@@ -0,0 +1,31 @@
+/// Panics and logs with error the `reason` (with file/line context.)
+pub fn log_panic(prefix: &str, provided_reason: impl AsRef<str>) {
+    let msg = format!("{}: '{}'", prefix, provided_reason.as_ref(),);
+    log::error!("{}", msg);
+    panic!("{}", msg);
+}
+
+pub fn incorrect_impl(reason: impl AsRef<str>) {
+    log_panic("Incorrect implementation", reason)
+}
+
+pub fn fatal_error(reason: impl AsRef<str>) {
```

Comment:
not yet heavily used, idea is to panic, but log with logger first, with predefined prefixes.